### PR TITLE
fix: rewrite test in direct comparison with strings

### LINKED ISSUE #56666: football team card project - step 27 missing space between name and (Captain)
### Describe the Issue

The code below is being accepted by step 27 of the football team card project (js beta) but in the next step, a space is added to the left of the `${name}` to correctly space the label (Captain) from the player's name.
It would be good if this space was explicitly requested (though I imagine it will cause trouble because more people will forget than add it correctly).
Or it would be good if the solution below was accepted as well as the actual solution as shown in step 28 (that is, if we don't want to bother explicitly asking for a space)

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-modern-javascript-methods-by-building-football-team-cards/step-27

### Your code

```
          <h2>${isCaptain ? '(Captain) ' : ''}${name}</h2>

```

### Expected behavior

the step should either ask for the space it requires or accept solutions that produce the same result

### Screenshots

_No response_

### System

N/A

### Additional context

forum post that led to this issue https://forum.freecodecamp.org/t/learn-modern-javascript-methods-by-building-football-team-cards-step-27/716079

### PR DESCRIPTION
**Pull request by hicugi**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56666

<hr>

Instead of regex match, I made a direct comparison with next solutions:

```js
const isTestOk =
  [
    "<h2>${isCaptain ? '(Captain) ' : ''}${name}</h2>",
    '<h2>${isCaptain ? "(Captain) " : ""}${name}</h2>',

    "<h2>${isCaptain == true ? '(Captain) ' : ''}${name}</h2>",
    "<h2>${isCaptain === true ? '(Captain) ' : ''}${name}</h2>",

    '<h2>${isCaptain == true ? "(Captain) " : ""}${name}</h2>',
    '<h2>${isCaptain === true ? "(Captain) " : ""}${name}</h2>',
  ].includes(title);
```

### CODE REVIEW

**Comment by ilenia-magoni** [comment]
I can write correct code and fail to pass this step, there is need for a test that checks the result and not the code written

**Comment by hicugi** [comment]
> I can write correct code and fail to pass this step, there is need for a test that checks the result and not the code written

Oh, my bad.
You are absolutely right, how about this solution?
Should work pretty fine with existing codebase:
```js
const players = [
  { name: "First", isCaptain: true },
  { name: "Second", isCaptain: false },
];

const before = playerCards.innerHTML;
setPlayerCards(players);
const after = playerCards.innerHTML.replace(before, "");

assert.ok(after.includes("<h2>(Captain) First</h2>"));
assert.ok(after.includes("<h2>Second</h2>"));
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md [hunk 1]
```diff
@@ -16,7 +16,17 @@ Before the `${name}` expression, add a new embedded expression. Inside that expr
...
 ```js
-assert.match(code, /\s*<h2\s*>\s*\$\{isCaptain(\s*===\s*true)?\s*\?\s*('|"|`)\(Captain\)\2\s*:\s*('|"|`)\3\}\s*\${\s*name\s*}\s*<\/h2>\s*/)
+const players = [
+  { name: "First", isCaptain: true },
+  { name: "Second", isCaptain: false },
+];
+
+const before = playerCards.innerHTML;
+setPlayerCards(players);
+const after = playerCards.innerHTML.replace(before, "");
```
[the remark this answers is not in the document]
Comment (hicugi) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md:26]: Sure, the idea is next:
- 24: I had this content
- 25: Added some players by function `setPlayerCards`
- 26: Looking for new players in changed content. We are going to ignore the previous prefilled list in the block

[the remark this answers is not in the document]
Comment (gikf) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md:26]: Easier would be just emptying the `playerCards` at the start of test.

There's however another change that will be needed first. :eyes: `map` in `setPlayerCards` doesn't actually return anything until step 33. Change happens between step 32 and 33.

[the remark this answers is not in the document]
Comment (ilenia-magoni) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md]: change happens quietly? that should not be, argh!

[the remark this answers is not in the document]
Comment (hicugi) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md]: > Why don't you playerCards.innerHTML = ''; instead?

24-26, line of code changes :)
Agree on clearing the list, submitted the change without replacing stuff.

[the remark this answers is not in the document]
Comment (ilenia-magoni) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md]: So, I guess the best way would be to have the map callback be written as implicit return from the start. Are you up to make this change @hicugi ?

[the remark this answers is not in the document]
Comment (hicugi) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md]: > There's however another change that will be needed first. 👀 map in setPlayerCards doesn't actually return anything until step 33. Change happens quietly between step 32 and 33.

Agree, but not sure that - this should be fixed in current branch...
@ilenia-magoni what do you think?

[the remark this answers is not in the document]
Comment (hicugi) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md]: > So, I guess the best way would be to have the map callback be written as implicit return from the start. Are you up to make this change @hicugi ?

@ilenia-magoni done

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md [hunk 2]
```diff
@@ -452,7 +461,7 @@ const setPlayerCards = (arr = players) => {
 
   playerCards.innerHTML += arr.map(
     ({ name, position, number, isCaptain, nickname }) => {
-      `
+      return `
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e95e39860dc5b01ebe9be0.md:464]: that's not implicit return. Also I meant updating from step 24 (instructions), up to step 32

**Comment by ilenia-magoni** [comment]
Thank you for your PR, but we are going to go with #56918

**Pull request closed without merging**